gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/QueryAndGatewayDtoConverter.java

```diff
@@ -342,60 +342,63 @@ private static Function convertToQueryFunction(
     // AVGRATE is adding a specific implementation because Pinot does not directly support this
     // function
     switch (function.getFunction()) {
-      case AVGRATE: {
-        builder.setFunctionName(FunctionType.SUM.name()).setAlias(function.getAlias());
-
-        // Adds only the argument that is a column identifier for now.
-        List<org.hypertrace.gateway.service.v1.common.Expression> columns =
-            function.getArgumentsList().stream()
-                .filter(org.hypertrace.gateway.service.v1.common.Expression::hasColumnIdentifier)
-                .collect(Collectors.toList());
-        columns.forEach(e -> builder.addArguments(convertToQueryExpression(e)));
-        break;
-      }
-      case PERCENTILE: {
-        org.hypertrace.gateway.service.v1.common.Expression percentileExp =
-            function.getArgumentsList().stream()
-                .filter(org.hypertrace.gateway.service.v1.common.Expression::hasLiteral)
-                .findFirst()
-                .orElseThrow(); // Should have validated arguments using PercentileValidator
-
-        long percentile = percentileExp.getLiteral().getValue().getLong();
-        String functionName = function.getFunction().name() + percentile;
-        builder.setFunctionName(functionName).setAlias(function.getAlias());
-
-        // Adds only the argument that is a literal identifier. This will bring the required nth
-        // percentile.
-        List<org.hypertrace.gateway.service.v1.common.Expression> columns =
-            function.getArgumentsList().stream()
-                .filter(org.hypertrace.gateway.service.v1.common.Expression::hasColumnIdentifier)
-                .collect(Collectors.toList());
-        columns.forEach(e -> builder.addArguments(convertToQueryExpression(e)));
-        break;
-      }
-      default: {
-        builder.setFunctionName(function.getFunction().name()).setAlias(function.getAlias());
-
-        if (function.getArgumentsCount() > 0) {
-          function
-              .getArgumentsList()
-              .forEach(
-                  e -> {
-                    // Health expressions are treated differently so ignore them.
-                    if (!e.hasHealth()) {
-                      builder.addArguments(convertToQueryExpression(e));
-                    }
-                  });
+      case AVGRATE:
```

Comment:
Is there any logic change here?

Reply:
In this file yes, but not in this function. All meaningful updates are line 419 and below (`hasTimeRangeFilter`, may have come from elsewhere, but I can't recall if I ended up updating it)